nixos/modules/services/databases/postgresql.nix

```diff
@@ -314,6 +314,14 @@ in
      "/share/postgresql"
     ];
 
+    systemd.tmpfiles.rules = (if groupAccessAvailable then [
+      "d '${cfg.dataDir}' 0750 postgres postgres - -"
+      "z '${cfg.dataDir}' 0750 postgres postgres - -"
+    ] else [
+      "d '${cfg.dataDir}' 0700 postgres postgres - -"
+      "z '${cfg.dataDir}' 0700 postgres postgres - -"
+    ]);
```

Comment:
From the `dataDir` description:

```
The data directory for PostgreSQL. If left as the default value
this directory will automatically be created before the PostgreSQL server starts, otherwise
the sysadmin is responsible for ensuring the directory exists with appropriate ownership
and permissions.
```

There are several reasons we shouldn't automatically provision the `dataDir` directory in this module, some of which include:
- current state isn't very reproducible when `dataDir` deviates from the expected value (`/var/lib/postgresql`)... also we don't have the ability to understand what the filesystem looks in order to make the right choices
  - maybe the user has network mounted drives we don't know about
  - maybe the user is using btrfs subvolumes
  - maybe something else you never considered
- modules make too many assumptions about `dataDir`, resulting in various permissions issues:
  - https://github.com/NixOS/nixpkgs/issues/80827
  - https://github.com/NixOS/nixpkgs/issues/81910
  - https://discourse.nixos.org/t/nixpkgs-policy-as-for-systemd-prestart-setup-scripts-vs-systemd-tmpfiles/5839/7
- when given arbitrary paths `systemd.tmpfiles.rules` generally doesn't behave as most users would expect
  - `[ "d /a foo foo" "d /a/b/c foo foo" ]` produces an error (rightfully so) because `/a/b` would be owned by `root:root`

If `dataDir` is left default then these rules are redundant and might cause problems. If the user specifies something other than the default then you're making too many assumptions. Remember that `tmpfile` rules are executed **very** early in system activation.

Please don't include these `tmpfiles` rules.

Reply:
Ok. Removed `systemd.tmpfiles.rules`.
In the mysql module in rules `systemd.tmpfiles.rules` i did not notice mistakes in work.